src/core/2d_primitives.js

```diff
@@ -233,21 +215,12 @@ p5.prototype.ellipse = function() {
  *
  */
 p5.prototype.line = function() {
-  if (!this._renderer._doStroke) {
-    return this;
-  }
-  var args = new Array(arguments.length);
-  for (var i = 0; i < args.length; ++i) {
-    args[i] = arguments[i];
-  }
+  p5._validateParameters('line', arguments);
```

Comment:
I think these call should generally only happen if the _doStroke and _doFill checks pass, to save time. What do you think?

Reply:
hmmm... you're right, the parameters aren't used if the renderer's current state doesn't match certain requirements - and this is true for several other methods throughout p5. but... i think if I were using this api and i was passing the wrong stuff to it, i'd want to know as soon as possible that i was making an error. if the renderer's state depends on some non-deterministic process and only triggers this validation error rarely then I'd have a Heisenbug on my hands :(

i believe that the perf implications of doing this every time should be reasonably minimal. especially in the minimized build, it's essentially a no-op.